analyzer/pubclient/pubclient_test.go

```diff
@@ -0,0 +1,95 @@
+package pubclient
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func wasteResp(resp *http.Response) error {
+	_, err := io.Copy(io.Discard, resp.Body)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func requireErrorAndWaste(t *testing.T, resp *http.Response, err error) {
+	if err == nil {
+		require.NoError(t, wasteResp(resp))
+	}
+	require.Error(t, err)
+}
+
+func requireNoErrorAndWaste(t *testing.T, resp *http.Response, err error) {
+	if err == nil {
+		require.NoError(t, wasteResp(resp))
+	}
+	require.NoError(t, err)
+}
+
+func TestMisc(t *testing.T) {
+	var requested bool
+	testServer := http.Server{
+		Addr: "127.0.0.1:8001",
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			requested = true
+		}),
+		ReadHeaderTimeout: 30 * time.Second,
+	}
+	serverErr := make(chan error)
+	go func() {
+		serverErr <- testServer.ListenAndServe()
+	}()
+	ctx := context.Background()
+
+	// Default client should reach local server. This makes sure the test server is working.
+	resp, err := getWithContextWithClient(ctx, http.DefaultClient, "http://localhost:8001/test.json")
+	requireNoErrorAndWaste(t, resp, err)
+	require.True(t, requested)
+	requested = false
+
+	// Hostname of test server
+	resp, err = GetWithContext(ctx, "http://localhost:8001/test.json")
+	requireErrorAndWaste(t, resp, err)
+	require.ErrorIs(t, err, NotPermittedError{})
+	require.False(t, requested)
+
+	// IP address of test server
+	resp, err = GetWithContext(ctx, "http://127.0.0.1:8001/test.json")
```

Comment:
I'm sure it behaves as expected, but maybe also include a test with a local literal IPv6 format, e.g. `http://[fe80::0202:b3ff:fe1e:8329]:8080/`

Reply:
ya let me try that

Reply:
added